### PR TITLE
Adding case resolution reason

### PR DESCRIPTION
**Pull request by kevgliss**

### CODE REVIEW

File: tests/factories.py [hunk 1]
```diff
@@ -727,6 +727,8 @@ class CaseFactory(BaseFactory):
     name = FuzzyText()
     title = FuzzyText()
     description = FuzzyText()
+    resolution = FuzzyText()
+    resolution_reason = FuzzyChoice(["False Positivie", "User Acknowleged"])
```
Comment (wssheldon) [inline tests/factories.py:731]:
```suggestion
    resolution_reason = FuzzyChoice(["False Positive", "User Acknowledged"])
```

File: src/dispatch/plugins/dispatch_slack/fields.py [hunk 2]
```diff
@@ -360,6 +362,28 @@ def resolution_input(
...
 
+def case_resolution_reason_select(
+    db_session: SessionLocal,
+    action_id: str = DefaultActionIds.case_resolution_reason_select,
+    block_id: str = DefaultBlockIds.case_resolution_reason_select,
+    label: str = "",
+    initial_option: dict = None,
+    **kwargs,
+):
+    """Creates an incident priority select."""
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/fields.py:373]: This docstring needs to be updated.

File: src/dispatch/static/dispatch/src/case/DetailsTab.vue [hunk 3]
```diff
@@ -29,6 +29,14 @@
           />
         </ValidationProvider>
       </v-flex>
+      <v-flex xs6>
+        <v-select
+          v-model="resolution_reason"
+          label="Resolution Reason"
+          :items="visibilities"
```
Comment (mvilanova) [inline src/dispatch/static/dispatch/src/case/DetailsTab.vue:36]: `resolutionReasons` instead?

**Review by mvilanova — CHANGES_REQUESTED** [review]

File: src/dispatch/static/dispatch/src/case/DetailsTab.vue (hunk 3 above)
Comment (kevgliss) [reply src/dispatch/static/dispatch/src/case/DetailsTab.vue:36]: That doesn't map to what the api returns.

Comment (mvilanova) [reply src/dispatch/static/dispatch/src/case/DetailsTab.vue:36]: I don't understand. You're providing visibilities not resolution reasons defined in https://github.com/Netflix/dispatch/pull/3090/files#diff-15e369cbda358d580577a6be233ec92377ef358ce0797560ff0f221e263fd47aR161 as an options to select from?

File: src/dispatch/plugins/dispatch_slack/fields.py (hunk 2 above)
Comment (mvilanova) [reply src/dispatch/plugins/dispatch_slack/fields.py:373]: Forgot to update it?

File: src/dispatch/static/dispatch/src/case/DetailsTab.vue (hunk 3 above)
Comment (mvilanova) [reply src/dispatch/static/dispatch/src/case/DetailsTab.vue:36]: Screenshot 2023-03-10 at 1 47 00 PM

**Pull request merged**